qemu/tests/blockdev_commit_server_down.py

```diff
@@ -0,0 +1,104 @@
+import time
+import socket
+
+from provider import job_utils
+from provider import backup_utils
+
+from provider.blockdev_commit_base import BlockDevCommitTest
+from provider.nbd_image_export import QemuNBDExportImage
+
+from virttest import env_process
+
+from avocado.core import exceptions
+
+
+class BlockdevCommitServerDown(BlockDevCommitTest):
+
+    def __init__(self, test, params, env):
+        params['nbd_export_format'] = params['image_format']
+        self.nbd_export = QemuNBDExportImage(params["local_image_tag"], params)
+        self.nbd_export.create_image()
+        self.nbd_export.export_image()
+
+        localhost = socket.gethostname()
+        params['nbd_server'] = localhost if localhost else 'localhost'
+        params['images'] += ' %s' % params['nbd_image_tag']
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        super(BlockdevCommitServerDown, self).__init__(test, params, env)
+
+    def check_commit_running(self):
+        tmo = int(self.params.get('commit_start_timeout', 5))
+
+        # make sure commit is running, i.e. offset > 0
+        for i in range(tmo):
+            time.sleep(1)
+            job = job_utils.get_block_job_by_id(self.main_vm,
+                                                self.commit_job)
+            if job['offset'] > 0:
+                break
+        else:
+            raise exceptions.TestFail("offset is 0 after %s seconds" % tmo)
```

Comment:
You could use ` test.fail("offset is 0 after %s seconds" % tmo)`

Reply:
done